library/alloc/src/collections/vec_deque/iter.rs

```diff
@@ -13,38 +15,44 @@ use super::{count, wrap_index, RingSlices};
 /// [`iter`]: super::VecDeque::iter
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct Iter<'a, T: 'a> {
-    ring: &'a [MaybeUninit<T>],
+    ring: *const [T],
     tail: usize,
     head: usize,
+    _marker: PhantomData<&'a T>,
 }
 
 impl<'a, T> Iter<'a, T> {
     pub(super) fn new(ring: &'a [MaybeUninit<T>], tail: usize, head: usize) -> Self {
-        Iter { ring, tail, head }
+        Iter {
+            ring: ptr::slice_from_raw_parts(ring.as_ptr().cast(), ring.len()),
+            tail,
+            head,
+            _marker: PhantomData,
+        }
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+unsafe impl<T: Sync> Sync for Iter<'_, T> {}
+#[stable(feature = "rust1", since = "1.0.0")]
+unsafe impl<T: Send> Send for Iter<'_, T> {}
```

Comment:
This doesn't seem to match the current implementation. Looks like we currently bound Send on T: Sync (Which seems correct, since this is handing out *shared* references).

https://doc.rust-lang.org/nightly/std/collections/vec_deque/struct.Iter.html#impl-Send-for-Iter%3C%27a%2C%20T%3E